thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -172,21 +176,23 @@ export default Ember.Controller.extend({
     const { selectedUrns, _pendingEntitiesRequests: pending, _entitiesCache: entitiesCache, _timeseriesCache: timeseriesCache } =
       this.getProperties('selectedUrns', '_pendingEntitiesRequests', '_entitiesCache', '_timeseriesCache');
 
-    // NOTE: potential race condition?
-
     // update pending requests
     pending.delete(framework);
 
     // timeseries eviction
     // TODO optimize for same time range reload
-    incoming.forEach(e => delete timeseriesCache[e.urn]);
+    Object.keys(incoming).forEach(urn => delete timeseriesCache[urn]);
```

Comment:
fyi, we can use a Map https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
which has the 2 following methods: 

#### Map.prototype.clear()
```
Removes all key/value pairs from the Map object.
```
#### Map.prototype.delete(key)
```
Returns true if an element in the Map object existed and has been removed, or false if the element does not exist. Map.prototype.has(key) will return false afterwards.
```

Reply:
yeah I saw that one too. if we introduce it, we should replace all plain JS objects with it for consistency